app/models/user.rb

```diff
@@ -36,6 +36,6 @@ def daily_habits
   end
 
   def daily_mood
-    moods.current_day_mood
+    moods.find_by(created_at: Date.current.all_day)
```

Comment:
Interestingly, moving this query to the current_day_mood scope didn't work. It returned an array of moods, but having the exact same line of code outside of the scope returned the expected nil instead of an array.

I left the current_day_mood scope untouched because it is used in testing.

Reply:
Interesting, I'll look into it some more.